cpp/src/randomforest/randomforest.cu

```diff
@@ -341,6 +343,126 @@ std::vector<unsigned char> save_model(ModelHandle model) {
+ *   each worker.
+ */
+void compare_concat_forest_to_subforests(
+  ModelHandle concat_tree_handle, std::vector<ModelHandle> treelite_handles) {
+  size_t concat_forest;
+  size_t total_num_trees = 0;
+  for (int forest_idx = 0; forest_idx < treelite_handles.size(); forest_idx++) {
+    size_t num_trees_each_forest;
+    TREELITE_CHECK(TreeliteQueryNumTree(treelite_handles[forest_idx],
+                                        &num_trees_each_forest));
+    total_num_trees = total_num_trees + num_trees_each_forest;
+  }
+
+  TREELITE_CHECK(TreeliteQueryNumTree(concat_tree_handle, &concat_forest));
+
+  ASSERT(
+    concat_forest == total_num_trees,
+    "Error! the number of trees in the concatenated forest and the sum "
+    "of the trees present in the forests present in each worker are not equal");
+
+  int concat_mod_tree_num = 0;
+  tl::Model& concat_model = *(tl::Model*)(concat_tree_handle);
+  for (int forest_idx = 0; forest_idx < treelite_handles.size(); forest_idx++) {
+    tl::Model& model = *(tl::Model*)(treelite_handles[forest_idx]);
+
+    ASSERT(
+      concat_model.num_feature == model.num_feature,
+      "Error! number of features mismatch between concatenated forest and the"
+      " individual forests ");
+    ASSERT(concat_model.num_output_group == model.num_output_group,
+           "Error! number of output group mismatch between concatenated forest "
+           "and the"
+           " individual forests ");
+    ASSERT(concat_model.random_forest_flag == model.random_forest_flag,
+           "Error! random forest flag value mismatch between concatenated "
+           "forest and the"
+           " individual forests ");
+
+    for (int indiv_trees = 0; indiv_trees < model.trees.size(); indiv_trees++) {
+      compare_trees(concat_model.trees[concat_mod_tree_num + indiv_trees],
+                    model.trees[indiv_trees]);
+    }
+    concat_mod_tree_num = concat_mod_tree_num + model.trees.size();
+  }
+}
+
+/**
+ * @brief Concatenates the forest information present in different workers to
+ *  create a single forest.
+ * @param[in] concat_tree_handle: ModelHandle for the concatenated forest.
+ * @param[in] treelite_handles: List containing ModelHandles for the forest present in
+ *   each worker.
+ */
+ModelHandle concatenate_trees(std::vector<ModelHandle> treelite_handles) {
+  tl::Model& first_model = *(tl::Model*)treelite_handles[0];
+  tl::Model* concat_model = new tl::Model;
+  for (int forest_idx = 0; forest_idx < treelite_handles.size(); forest_idx++) {
+    tl::Model& model = *(tl::Model*)treelite_handles[forest_idx];
+    concat_model->trees.insert(concat_model->trees.end(), model.trees.begin(),
+                               model.trees.end());
+  }
+  concat_model->num_feature = first_model.num_feature;
+  concat_model->num_output_group = first_model.num_output_group;
+  concat_model->random_forest_flag = first_model.random_forest_flag;
```

Comment:
Does anything check that these models are consistent in terms of number of output groups, num_features, etc? I think you used to have that call to check but took it out.

Reply:
The number of features for all is obtained from the number of columns of the dataset, the num_output_group and random_forest_flag variables are set while converting cuML RF to treetlite. These values gets checked for all the different models in the main check function `compare_concat_forest_to_subforests`